pkgs/applications/display-managers/greetd/qtgreet/default.nix

```diff
@@ -0,0 +1,58 @@
+{
+  stdenv,
+  lib,
+
+  fetchFromGitLab,
+
+  wrapQtAppsHook,
+  meson,
+  pkg-config,
+  cmake,
+  qttools,
+  ninja,
+
+  qtbase,
+  json_c,
+  wlroots,
+  wayqt,
+  wayland,
+  pixman
+}:
```

Comment:
Alejandra shouldn't be used in nixpkgs.

Reply:
And style shouldn't be mandated in Nixpkgs until it's codified.

Once we have decided on a style in an RFC, I will use that. Until then, let me use whatever I want.

Reply:
That is not how it works or could work. If we don't do anything until it is fully automated then we don't get anywhere.

Alejandra purposely decided to do all formatting differently than nixpkgs, nix-fmt or nixpkgs-fmt and we can't use two formatters that are complete opposites.

Reply:
I'm not talking about automation, I'm talking about preference. Until we have decided on one set of answers to the stylistic options that do exist, there is no point in *mandating* one option over the other.

You can absolutely express your subjective preferences but it's a nitpick at most and not at all a reason to block a PR.

If I was touching existing code, I would continue its style and not suddenly use a different one or reformat everything for a few lines of my own. I'd be fully with you if that were the case.  
This is entirely new code however.

Alejandra evaluated style from scratch without letting existing code influence the decision (much). Where it does things differently than what's often used in Nixpkgs, that's usually for good reason, not to be contra.  
I once held a strong preference for leading commas in function args too but trailing comma style has objective upsides and I think those are more important than my subjective preference.

PS: I don't actually use Alejandra.

Reply:
> that's usually for good reason

How can you even _know_ there are good reasons at all, whereas your definitive argument is "Formatting is subjective"?

Reply:
Some aspects of code style are not subjective. Diff noise for example (when you need to update unrelated lines to do your change somewhere else, adding a comma to the previous line for example). I'm sure many others exist.  
My argument isn't that Alejandra's style is somehow objectively perfect or anything like that, it's that thought was put into it and it's not merely different for the sake of being different as was claimed.

My "definitive argument" is not that you should let me do whatever I want, it's that there is no commonly-agreed style yet and therefore you shouldn't be mandating any.  
If you want to enforce a style (which I'd be 100% in support of), it needs to be defined and agreed upon first. RFCs are the place to do that however, not PR reviews.

Reply:
> it's that there is no commonly-agreed style yet and therefore you shouldn't be mandating any.

The commonly agreed style is the de facto coding standard which uses leading commas.

Reply:
> The commonly agreed style

Could you point me to the official place where we "commonly agreed" on the style?